wgpu-core/src/device/global.rs

```diff
@@ -1377,6 +1377,7 @@ impl Global {
             .command_buffers
             .unregister(command_encoder_id.transmute())
         {
+            cmd_buf.data.lock().as_mut().unwrap().encoder.discard();
```

Comment:
This `wgpu-core`-centric change might be relevant to the issue that #4023 was trying to solve in Metal; perhaps that fix is no longer necessary?

- [ ] Investigate.

Reply:
This also means that the test I had added before is redundant, woot!

Reply:
Thinking about it, I think the appropriate solution is actually to have every encoder for a specific back end discard itself on drop, since that makes sense for them. We could also have the core abstraction discard the encoding, since that's not incorrect. It would be redundant, technically, but I don't see that as a bad thing.

Reply:
Also added a `Drop` impl. for GLES.

It seems that Vulkan specifically does not appear to track its current state, relying on correct usage from `wgpu-core`. I won't add a `Drop` call to `discard_encoding` for that backend.

Reply:
Added a follow-up for Vulkan at <https://github.com/gfx-rs/wgpu/issues/5255> .